src/services/jsTyping.ts

```diff
@@ -53,6 +53,7 @@ namespace ts.JsTyping {
      */
     export function discoverTypings(
         host: TypingResolutionHost,
+        log: ((message: string) => void) | undefined,

```

Comment:
We're doing this kind of pattern (`if (this.log.isEnabled()) this.log.writeLine(...)`) for logging in `typingsInstaller`, I think for performance so that we don't allocate strings that we won't need. It may not matter compared to all the I/O we're doing though.